CIScripts/Tests/VRouterAgentTests.ps1

```diff
@@ -0,0 +1,150 @@
+        Initialize-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+        Assert-ExtensionIsRunning -Session $Session -TestConfiguration $TestConfiguration
+
+        # Given clean vRouter
+        Assert-NoVifs -Session $Session
+
+        # When Agent is started
+        Enable-VRouterAgent -Session $Session
+        Assert-AgentIsRunning -Session $Session
+        Start-Sleep -Seconds 15
+
+        # Then pkt0 appears in vRouter
+        Assert-IsPkt0Injected -Session $Session
+    }
+
+    function Test-Pkt0RemainsInjectedAfterAgentStops {
+        Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
+               [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
+
+        # Given Extension is running
+        Clear-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+        Initialize-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+        Assert-ExtensionIsRunning -Session $Session -TestConfiguration $TestConfiguration
+
+        # Given Agent is running
+        Enable-VRouterAgent -Session $Session
+        Assert-AgentIsRunning -Session $Session
+        Start-Sleep -Seconds 15
+
+        # Given pkt0 is injected
+        Test-IsPkt0Injected -Session $Session
+
+        # When Agent is stopped
+        Disable-VRouterAgent -Session $Session
+        Assert-AgentIsNotRunning -Session $Session
+
+        # Then pkt0 exists in vRouter
+        Assert-IsPkt0Injected -Session $Session
+    }
+
+    function Test-OnePkt0ExistsAfterAgentIsRestarted {
+        Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
+               [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
+
+        # Given Extension is running
+        Clear-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+        Initialize-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+
+        # Given Agent is running
+        Enable-VRouterAgent -Session $Session
+        Test-IsVRouterAgentEnable -Session $Session
+        Start-Sleep -Seconds 15
+
+        # Given pkt0 is injected
+        Test-IsPkt0Injected -Session $Session
+
+        # When Agent is restarted
+        Disable-VRouterAgent -Session $Session
+        Assert-AgentIsNotRunning -Session $Session
+        Enable-VRouterAgent -Session $Session
+        Assert-AgentIsRunning -Session $Session
+        Start-Sleep -Seconds 15
+
+        # Then pkt0 exists in vRouter
+        Assert-IsOnlyOnePkt0Injected -Session $Session
+    }
+
+    Test-InitialPkt0Injection -Session $Session -TestConfiguration $TestConfiguration
+    Test-Pkt0RemainsInjectedAfterAgentStops -Session $Session -TestConfiguration $TestConfiguration
+    Test-OnePkt0ExistsAfterAgentIsRestarted -Session $Session -TestConfiguration $TestConfiguration
```

Comment:
Please add `Clear-TestConfiguration` at the end of the test. ` TestIntegration.ps1` assumes that everything is cleaned-up after each test.